### DISCUSSION TITLE
Nunit Console Runner - Failed to Resolve Assembly

### DISCUSSION THREAD

**Discussion by binosheen97** [nunit/nunit-console#1399] [Q&A]
Hi,

In my C# .Net Framework v4.8 project, I have installed few packages including Nunit console runner and Dapper. When i build i am not getting any error but in jenkins while intrumenting i am getting the below error. If i remove any one of this packages, then i am not getting this error. Please help me to resolve this.

ERROR *** Instrumentation phase failed
Failed to resolve assembly: 'System.Data.Common, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'

**Comment by mikkelbu** [comment]
From this SteveGilham/altcover#187 it sounds like this could be related to Mono.Cecil.

Which version of NUnit and NUnit.Console are you using? and is there more information in log flies etc.

**Comment by binosheen97** [comment]
package id="NUnit" version="3.13.2" targetFramework="net48"
  package id="NUnit.ConsoleRunner" version="3.12.0" targetFramework="net48"
  package id="NUnit3TestAdapter" version="4.1.0" targetFramework="net48"

**Comment by OsirisTerje** [comment]
Mono.Cecil is not used in 4.1/3.12, so it must be something different. 

Can you please upload a repro project ?

**Comment by binosheen97** [comment]
Additional log Information given below. It has resolved assembly reference System.Data from the below path but its failed to resolve System.Data.Common.
/usr/share/dotnet/shared/Microsoft.NETCore.App/5.0.7/System.Data.dll. (Refer snap)

Log:-

/home/jenkins/agent/workspace/l_AETool_feature_US191919/Tool_AETool/bin/Debug/__Instrumented/NUnit3.TestAdapter.dll
                <=  NUnit3.TestAdapter, Version=4.1.0.0, Culture=neutral, PublicKeyToken=null
/home/jenkins/agent/workspace/l_AETool_feature_US191919/Tool_AETool/bin/Debug/__Instrumented/Dapper.dll
                <=  Dapper, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null
    
Resolved assembly reference 'System.Data, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' as file '/usr/share/dotnet/shared/Microsoft.NETCore.App/5.0.7/System.Data.dll'.

ERROR *** Instrumentation phase failed
Failed to resolve assembly: 'System.Data.Common, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'

![image](https://github.com/nunit/nunit-console/assets/113877906/54c10014-79ee-44d4-bda9-3973b819a957)

**Comment by binosheen97** [comment]
/usr/share/dotnet/shared/Microsoft.NETCore.App/5.0.7/System.Data.dll

This dll (version 4.0.0.0) has assembly reference System.Data.Common (version 0.0.0.0).
But when i check the System.Data.Common.dll directly from the below path, it is showing the version as 5.0.0.0.
/usr/share/dotnet/shared/Microsoft.NETCore.App/5.0.7/System.Data.Common.dll

Is that the issue?

**Comment by OsirisTerje** [comment]
Hard to say.  A repro would help.
